indexd/src/quic/client.rs

```diff
@@ -138,10 +138,15 @@ impl Client {
         let mut hosts_map = self.inner.hosts.lock().unwrap();
         let mut priority_queue = self.inner.preferred_hosts.lock().unwrap();
         hosts_map.clear();
-        for host in hosts {
+        for (idx, host) in hosts.into_iter().enumerate() {
             hosts_map.insert(host.public_key, host.addresses);
+            let bias = -(idx as i64);
             if !priority_queue.contains(&host.public_key) {
-                priority_queue.push(host.public_key, HostMetric::default());
+                priority_queue.push(host.public_key, HostMetric::with_bias(bias));
+            } else {
+                priority_queue.change_priority_by(&host.public_key, |metric| {
```

Comment:
I don't think you need this. You should be able to just remove the whole `if/else` and `.push` all hosts into the queue. Because `push` handles duplicate entries correctly and updates their priority.